pkg/instancetype/apply/cpu_test.go

```diff
@@ -0,0 +1,605 @@
+//nolint:dupl
+package apply_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	v1 "kubevirt.io/api/core/v1"
+	instancetypev1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	"kubevirt.io/kubevirt/pkg/instancetype/apply"
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/pointer"
+)
+
+var _ = Describe("Apply to vmi - cpu", func() {
+	var (
+		vmi              *v1.VirtualMachineInstance
+		instancetypeSpec *instancetypev1beta1.VirtualMachineInstancetypeSpec
+		preferenceSpec   *instancetypev1beta1.VirtualMachinePreferenceSpec
+
+		vmiApplier = apply.NewVMIApplier()
+		field      = k8sfield.NewPath("spec", "template", "spec")
+	)
+
+	BeforeEach(func() {
+		vmi = libvmi.New()
+
+		instancetypeSpec = &instancetypev1beta1.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1beta1.CPUInstancetype{
+				Guest:                 uint32(2),
+				Model:                 pointer.P("host-passthrough"),
+				DedicatedCPUPlacement: pointer.P(true),
+				IsolateEmulatorThread: pointer.P(true),
+				NUMA: &v1.NUMA{
+					GuestMappingPassthrough: &v1.NUMAGuestMappingPassthrough{},
+				},
+				Realtime: &v1.Realtime{
+					Mask: "0-3,^1",
+				},
+				MaxSockets: pointer.P(uint32(6)),
+			},
+		}
+
+		preferenceSpec = &instancetypev1beta1.VirtualMachinePreferenceSpec{
+			CPU: &instancetypev1beta1.CPUPreferences{},
+		}
+	})
+
+	It("should default to PreferSockets", func() {
+		Expect(vmiApplier.ApplyToVMI(field, instancetypeSpec, preferenceSpec, &vmi.Spec, &vmi.ObjectMeta)).ToNot(HaveOccurred())
```

Comment:
nit - Apologies if I suggested `ToNot(HaveOccured())` before but `To(Succeed())` would actually be better IMHO.